### PR TITLE
Introduce "setstats" option in gdal_edit.py

### PR DESCRIPTION
**Pull request by swalterfub**
With "-setstats" we can "fake" statistics which is very useful for tricking GIS programs to set the symbology as we want to. Can also eliminate outliers, etc...

### CODE REVIEW

**Comment by rouault** [comment]
Can you document the new option in swig/python/scripts/gdal_edit.dox ?

**Comment by swalterfub** [comment]
It's documented now and extended, so that any parameter can  be set to -1 (is there a better non-existing value?) and the ones set to -1 get calculated from statistics.

**Comment by rouault** [comment]
Ah one point I've missed in my intial review is that there's no test for this functionality. Can you add one in autotest/pyscripts/test_gdal_edit.py ?

**Comment by swalterfub** [comment]
Test included.

File: autotest/pyscripts/test_gdal_edit.py [hunk 1]
```diff
@@ -309,7 +367,7 @@ def test_gdal_edit_py_6():
 # Test -colorinterp_X
 
 
-def test_gdal_edit_py_7():
+def test_gdal_edit_py_8():
```
Comment (rouault) [inline autotest/pyscripts/test_gdal_edit.py]: test_gdal_edit_py_8 must be added to the gdaltest_list array towards bottom of the file

Comment (swalterfub) [reply autotest/pyscripts/test_gdal_edit.py]: Done. Thanks!

File: autotest/pyscripts/test_gdal_edit.py [hunk 2]
```diff
@@ -293,6 +293,64 @@ def test_gdal_edit_py_6():
...
+    if stat_mean is None or float(stat_mean) != 100:
+        gdaltest.post_reason('fail')
+        return 'fail'
+    stat_stddev = ds.GetRasterBand(1).GetMetadataItem('STATISTICS_STDDEV')
+    if stat_stddev is None or float(stat_stddev) != 30:
+        gdaltest.post_reason('fail')
+        return 'fail'
+    ds = None
+    
+    print(stat_min,stat_max,stat_mean,stat_stddev)
```
Comment (rouault) [inline autotest/pyscripts/test_gdal_edit.py]: could you remove this debug trace ?

**Pull request merged**